scripts/update_capacity_configuration.py

```diff
@@ -0,0 +1,220 @@
+    """
+    existing_capacity_modes = [mode for mode in data if mode in capacity_config]
+    updated_capacity_config = deepcopy(capacity_config)
+    for mode in existing_capacity_modes:
+        if isinstance(capacity_config[mode], float | int):
+            updated_capacity_config[mode] = data[mode]
+        elif isinstance(capacity_config[mode], dict):
+            updated_capacity_config[mode] = get_zone_capacity_dict(
+                mode, capacity_config, data
+            )
+        elif isinstance(capacity_config[mode], list):
+            updated_capacity_config[mode] = get_zone_capacity_list(
+                mode, capacity_config, data
+            )
+
+    new_modes = [m for m in data if m not in capacity_config]
+    for mode in new_modes:
+        updated_capacity_config[mode] = data[mode]
+    return updated_capacity_config
+
+
+def get_zone_capacity_dict(
+    mode: str, capacity_config: dict[str, Any], data: [str, Any]
+) -> dict[str, Any]:
+    """Generate the updated capacity config for a zone if the capacity config is a dict"""
+    existing_capacity = capacity_config[mode]
+    if existing_capacity["datetime"] != data[mode]["datetime"]:
+        return [existing_capacity] + [data[mode]]
+    else:
+        return data[mode]
+
+
+def get_zone_capacity_list(
+    mode: str, capacity_config: dict[str, Any], data: [str, Any]
+) -> list[dict[str, Any]]:
+    """Generate the updated capacity config for a zone if the capacity config is a list"""
+    if data[mode]["datetime"] not in [d["datetime"] for d in capacity_config[mode]]:
+        return capacity_config[mode] + [data[mode]]
+    else:
+        # replace the capacity data point with the same datetime by the most recent update (from data)
+        return [
+            data[mode] if item["datetime"] == data[mode]["datetime"] else item
+            for item in capacity_config[mode]
+        ]
+
+
+def check_capacity_config_type(capacity_config: list, config_type: type) -> None:
+    """Check that the capacity config is of the specified type"""
+    return (
+        True
+        if all(isinstance(item, config_type) for item in capacity_config)
+        else False
+    )
+
+
+def get_aggregated_capacity_config(
+    parent_zone: ZoneKey,
+) -> dict[str, Any] | None:
+    """Generate the aggregated capacity config for a parent zone"""
+    zone_keys = [zone for zone in ZONE_PARENT if ZONE_PARENT[zone] == parent_zone]
+    parent_capacity_config = ZONES_CONFIG[parent_zone]["capacity"]
+
+    all_capacity_configs = [ZONES_CONFIG[zone]["capacity"] for zone in zone_keys]
+
+    for mode in CAPACITY_MODES:
+        mode_capacity_configs = [
+            capacity_config[mode] for capacity_config in all_capacity_configs
+        ]
+
+        if not check_capacity_config_type(
+            mode_capacity_configs, type(mode_capacity_configs[0])
+        ):
+            logger.warning(
```

Comment:
I think it would be better to raise. So that the person first fixes all child before.